rollout/canary.go

```diff
@@ -64,7 +64,12 @@ func (c *RolloutController) rolloutCanary(rollout *v1alpha1.Rollout, rsList []*a
 	}
 
 	logCtx.Info("Reconciling AnalysisRun step")
-	if err = c.reconcileAnalysisRuns(roCtx); err != nil {
+	addPause, err := c.reconcileAnalysisRuns(roCtx)
+	if addPause {
```

Comment:
Rather than surfacing a bool here, could we call

```golang
err := c.reconcileAnalysisRuns(roCtx)
if roCtx.PauseContext().HasAddPause() {
    logCtx.Info("Detected pause due to inconclusive AnalysisRun")
    return c.syncRolloutStatusCanary(roCtx)
}
if err != nil {
    return err
}
```